packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -702,6 +702,21 @@ abstract class SchedulerBinding extends BindingBase with ServicesBinding {
     });
   }
 
+  /// Similar to [scheduleWarmUpFrame] except it locks all incoming events from
+  /// flushing until the frame finished asynchronously.
+  void lockAndScheduleWarmUpFrame() {
+    Timeline.startSync('Warm-up frame');
+    final Completer<Null> frameCompleter = new Completer<Null>();
+    lockEvents(() => frameCompleter.future);
+
+    scheduleWarmUpFrame();
+
+    endOfFrame.then((_) {
+      frameCompleter.complete();
+      Timeline.finishSync();
+    });
+  }
```

Comment:
i think this would be clearer like this:

```dart
    Timeline.startSync('Warm-up frame');
    lockEvents(() async {
      scheduleWarmUpFrame();
      await endOfFrame;
      Timeline.finishSync();
    });
 ```

Reply:
Tried it initially, though the original semantic was for this method to complete after the frame is scheduled, not after the lock is set. (https://github.com/flutter/flutter/blob/master/packages/flutter/test/scheduler/scheduler_test.dart#L111)

Reply:
Oh, interesting.

Reply:
How about:

```dart
    Timeline.startSync('Warm-up frame');
    scheduleWarmUpFrame();
    lockEvents(() async {
      await endOfFrame;
      Timeline.finishSync();
    });
 ```